### PR TITLE
Fix issue with multiple result sets

### PR DESCRIPTION
**Pull request by abist**
There were a bunch of race conditions in the existing code, so fixed them while fixing this issue. Cleaned code to use `async`'s instead of `Thenables`. If you want me to put those changes in a different PR, let me know. It was confusing to read when fixing the race conditions so I changed it as a part of this PR.

### CODE REVIEW

File: src/controllers/connectionManager.ts [hunk 1]
```diff
@@ -99,6 +101,7 @@ export default class ConnectionManager {
         this._connections = {};
         this._connectionCredentialsToServerInfoMap =
             new Map<Interfaces.IConnectionCredentials, ConnectionContracts.ServerInfo>();
+        this._uriToConnectionPromiseMap = new Map<string, Deferred<boolean>>();
```
Comment (Charles-Gagnon) [inline src/controllers/connectionManager.ts]: nit: but you could just create this above when declaring the var

File: src/controllers/untitledSqlDocumentService.ts [hunk 2]
```diff
@@ -18,25 +18,16 @@ export default class UntitledSqlDocumentService {
...
+    public async newQuery(content?: string): Promise<vscode.TextEditor> {
+        // Open an untitled document. So the  file doesn't have to exist in disk
+        let doc = await this.vscodeWrapper.openMsSqlTextDocument(content);
+        // Show the new untitled document in the editor's first tab and change the focus to it.
+        const editor = await this.vscodeWrapper.showTextDocument(doc, 1, false);
+        const position = editor.selection.active;
+        let newPosition = position.with(position.line + 1, 0);
+        let newSelection = new vscode.Selection(newPosition, newPosition);
+        editor.selection = newSelection;
+        return Promise.resolve(editor);
```
Comment (Charles-Gagnon) [inline src/controllers/untitledSqlDocumentService.ts]: You can just return the editor here, no need to wrap in the Promise.resolve

File: src/models/connectionStore.ts [hunk 3]
```diff
@@ -279,18 +279,13 @@ export class ConnectionStore {
...
-                resolve(undefined);
-            }, err => {
-                reject(err);
-            });
-        });
+    public async clearRecentlyUsed(): Promise<void> {
+        // Update the MRU list to be empty
+        try {
+            await this._context.globalState.update(Constants.configRecentConnections, []);
+        } catch (error) {
```
Comment (Charles-Gagnon) [inline src/models/connectionStore.ts]: The try/catch is unnecessary here with an async function- the returned promise will be rejected if an exception is thrown already.

**Comment by abist** [comment]
@Charles-Gagnon We don't a want very big changes at the last minute, so I'm going to revert the `async`s to `Thenables` and have 2 PRs: one for the fix, and one cleaning stuff. I'm going to close this PR for now.

**Pull request closed without merging**